scripts/scr_heimao_step.gml

```diff
@@ -42,14 +42,21 @@ if(sitRelease)
 
 (inventory).active = keyInventory;
 
+// Check if heimao should be interrupted
+// Add on here to stop updating 
+shouldBeInterrupted = (inventory).active;
+
 /********************/ 
 /********************/ 
 
-// Update states accordingly
-scr_heimao_updateState();
-
-// Act
-scr_heimao_action();
-
-// Draw
-scr_heimao_draw();
+if (!shouldBeInterrupted)
```

Comment:
I know it's not your code, but the comments in this section aren't really useful tbh, you can delete them. the functions more or less express the intent.

Reply:
tfw it is :(